tripleblindmarket/private-kit
# SAF - 135 download over wifi only - part 1

#### Description:

Add the ability to only download HA data over wifi only if the user chooses to do so.

Changes include:
- Add the redux state to store app settings. Add the setting for downloading data over wifi only.
- In the Intersect code, before downloading the page data, check the added setting and connection type. If the user only wants to download over wifi and the current connection isn't wifi, skip the download of the page data.
- Add caching of page data so previously downloaded page can be used for intersection check.
- Add reactotron (only during development) for better network monitoring. Install the reactotron desktop app to use here https://github.com/infinitered/reactotron. Note, if you're running the app on an android device, besides reverse proxying port 8081, you'll also need to reverse proxy port 9090 by running the command ` adb -s android-device-id reverse tcp:9090 tcp:9090` where android device id is the id of the phone, you can find this by the command `adb devices` (while having the android device plugged in).

#### Linked issues:

https://pathcheck.atlassian.net/browse/SAF-135

#### Screenshots:

No screenshots yet as there isn't any UI available at this point. Currently waiting on design for this.

#### How to test:

1. By default, the app always allows downloading of page data. You'll see the page data request in reactotron as you change the selected HA. This should only happen once as the page data would be cached.
1. You can update the cache key to force cache miss, triggering the download again. 
1. Use redux dev tool to dispatch the action below to simulate wifi only mode.
```js
{
type: 'TOGGLE_DOWNLOAD_HA_DATA_OVER_WIFI_ONLY'
}
```
4. Turn the wifi off on the device. The download will no longer happen.
